scripts/validation/providers/validate.ts

```diff
@@ -0,0 +1,129 @@
+    const mdx = path.join(docsPath, `${filename}.mdx`);
+    const svg = path.join(svgPath, `${providerKey}.svg`);
+
+    if (!fs.existsSync(mdx)) {
+        console.error(chalk.red('error'), chalk.blue(providerKey), `Documentation file not found`);
+        console.error(`Expected file: ${mdx}`);
+        error = true;
+    }
+    if (!fs.existsSync(svg)) {
+        console.error(chalk.red('error'), chalk.blue(providerKey), `SVG file not found`);
+        console.error(`Expected file: ${svg}`);
+        error = true;
+    }
+
+    // Find all connectionConfig references
+    const connectionConfigReferences = findConnectionConfigReferences(provider);
+
+    // Check if referenced connectionConfig keys exist in the connection_config property
+    if (connectionConfigReferences.length > 0) {
+        for (const reference of connectionConfigReferences) {
+            const defined = provider.connection_config && reference.key in provider.connection_config;
+            const inTokenResponseMetadata = provider.token_response_metadata?.includes(reference.key);
+
+            if (!defined && !inTokenResponseMetadata) {
+                console.error(
+                    chalk.red('error'),
+                    chalk.blue(providerKey),
+                    `"${reference.path.join('" > "')}" use "connectionConfig.${reference.key}", but it's not defined in "connection_config"`
+                );
+                error = true;
+                continue;
+            }
+        }
+    } else if (provider.connection_config) {
+        console.error(chalk.red('error'), chalk.blue(providerKey), `"connection_config" is defined but not required`);
+        error = true;
+    }
+
+    if (provider.auth_mode === 'API_KEY') {
+        if (!provider?.credentials?.['apiKey']) {
+            console.error(chalk.red('error'), chalk.blue(providerKey), `"credentials" > "apiKey" is not defined`);
+            error = true;
+        }
+    } else {
+        if (provider.credentials) {
+            console.error(chalk.red('error'), chalk.blue(providerKey), `"credentials" is defined but not required`);
+            error = true;
+        }
+    }
+}
+
+// Function to recursively search for connectionConfig in the provider value
+interface Ref {
+    path: string[];
+    key: string;
+}
+function findConnectionConfigReferences(obj: Record<string, any>, path: string[] = []): Ref[] {
+    let references: Ref[] = [];
+    if (!(typeof obj === 'object') || !obj) {
+        return references;
+    }
+
+    for (const [key, value] of Object.entries(obj)) {
+        if (typeof value === 'string') {
+            const matches = value.matchAll(/\$\{connectionConfig\.([^}]+)\}/g);
+            for (const match of matches) {
+                references.push({ path: [...path, key], key: match[1]! });
+            }
+        } else if (typeof value === 'object' && value) {
+            references = references.concat(findConnectionConfigReferences(value, [...path, key]));
```

Comment:
no need for `concat` I think:
```
references.push(...findConnectionConfigReferences(value, [...path, key]));
```

Reply:
indeed (not going to lie it's chatgpt :whistle:)